gtda/externals/python/cubical_complex_interface.py

```diff
@@ -95,15 +95,14 @@ def persistence(self, homology_coeff_field=11, min_persistence=0):
         if self.pcohptr is not None:
             del self.pcohptr
         if self.thisptr is not None:
-            pass
             self.pcohptr = Cubical_complex_persistence_interface(self.thisptr,
                                                                  True)
-        persistence_result = []
+
         if self.pcohptr is not None:
-            persistence_result = self.pcohptr.get_persistence(
-                homology_coeff_field, min_persistence)
+            self.pcohptr.compute_persistence(homology_coeff_field,
+                                             min_persistence)
 
-        return persistence_result
+        return self.pcohptr.get_persistence()
```

Comment:
I think that an exception is thrown here if `self.pcohptr` is `None`. That's why an empty array was created before.

Reply:
Oh yeah sorry, you're right! Thanks for pointing this out, I'll put the empty lists back.

Reply:
Done!